MekHQ/src/mekhq/campaign/autoresolve/acar/handler/WithdrawActionHandler.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2024 - The MegaMek Team. All Rights Reserved.
+ *
+ *  This file is part of MekHQ.
+ *
+ *  MekHQ is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  MekHQ is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with MekHQ. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package mekhq.campaign.autoresolve.acar.handler;
+
+import megamek.common.Compute;
+import megamek.common.IEntityRemovalConditions;
+import mekhq.campaign.autoresolve.acar.SimulationManager;
+import mekhq.campaign.autoresolve.acar.action.WithdrawAction;
+import mekhq.campaign.autoresolve.acar.action.WithdrawToHitData;
+import mekhq.campaign.autoresolve.acar.report.WithdrawReporter;
+import mekhq.campaign.unit.damage.DamageApplierChooser;
+
+public class WithdrawActionHandler extends AbstractActionHandler {
+
+    private final WithdrawReporter reporter;
+
+    public WithdrawActionHandler(WithdrawAction action, SimulationManager gameManager) {
+        super(action, gameManager);
+        this.reporter = new WithdrawReporter(gameManager.getGame(), this::addReport);
+    }
+
+
+    @Override
+    public boolean cares() {
+        return game().getPhase().isEnd();
+    }
+
+    /**
+     * This is not up to rules as written, the intention was to create a play experience that is more challenging and engaging.
+     * The rules as written allow for a very simple withdraw mechanic that in this situation is very easy to exploit and would
+     * create too many games which result in no losses.
+     */
+    @Override
+    public void execute() {
+        var withdraw = (WithdrawAction) getAction();
```

Comment:
This is coming back to my earlier comments of using `var` v. specific type. I am really nervous with just how much you're using `var` throughout this module.

Reply:
<img width="658" alt="Screenshot 2024-12-16 at 19 44 56" src="https://github.com/user-attachments/assets/bc9ded51-57e2-4a36-9b9f-f9e46fedab9d" />
<img width="605" alt="Screenshot 2024-12-16 at 19 45 17" src="https://github.com/user-attachments/assets/92f567d1-8bff-4852-96d5-8a215fcab334" />
<img width="556" alt="Screenshot 2024-12-16 at 19 45 33" src="https://github.com/user-attachments/assets/7c04d77b-2525-4605-9fd2-a1f262c1d38f" />
<img width="750" alt="Screenshot 2024-12-16 at 19 45 48" src="https://github.com/user-attachments/assets/e6c58824-fc1e-43a2-87c0-a55b1cbcfa93" />

There is no need for using anything other than var inside a function to be honest with you.